src/sys/ptrace/linux.rs

```diff
@@ -481,3 +481,24 @@ pub unsafe fn write(
 {
     ptrace_other(Request::PTRACE_POKEDATA, pid, addr, data).map(drop)
 }
+
+/// Reads a word from a user area at `offset`.
+/// The user struct definition can be found in `/usr/include/sys/user.h`.
+pub fn read_user(pid: Pid, offset: AddressType) -> Result<c_long> {
+    ptrace_peek(Request::PTRACE_PEEKUSER, pid, offset, ptr::null_mut())
+}
```

Comment:
@asomers what are your thoughts on having separate `read` and `read_user` functions (and the same for write), vs. combining them and taking a third parameter for "data" or "user"?

Same for write.

Reply:
In this case I think it's better to have separate functions.  IIUC the PEEKDATA and PEEKUSER don't really do the same thing at all, and it's unlikely that a consumer's crate would have a single code path for both.